src/main/java/seedu/address/logic/parser/DeleteStudentCommandParser.java

```diff
@@ -1,31 +1,44 @@
 package seedu.address.logic.parser;
 
 import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_NUMBER;
+
+import java.util.stream.Stream;
 
 import seedu.address.logic.commands.DeleteStudentCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.person.Name;
+import seedu.address.model.student.StudentNumber;
 
 /**
  * Parses input arguments and creates a new DeleteStudentCommand object
  */
 public class DeleteStudentCommandParser implements Parser<DeleteStudentCommand> {
     @Override
     public DeleteStudentCommand parse(String userInput) throws ParseException {
-        String trimmedArgs = userInput.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteStudentCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(userInput,
+                PREFIX_NAME, PREFIX_STUDENT_NUMBER
+        );
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    DeleteStudentCommand.MESSAGE_USAGE));
         }
 
-        Name nameToDelete;
+        Name nameToDelete = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+
+        StudentNumber studentNumberToDelete;
```

Comment:
Perhaps you could just initialize it in the block below since this variable is not used by any lines after the block.